keras2onnx/parser.py

```diff
@@ -581,6 +585,14 @@ def _parse_graph_core(graph, keras_node_dict, topology, top_scope, output_names)
         nodes = []
         layer_key_, model_ = _parse_nodes(graph, inference_nodeset, input_nodes, keras_node_dict, keras_nodeset,
                                           node, nodes, varset, visited, q_overall)
+
+        # Only parse Keras layers once (allow certain shared classes)
+        if layer_key_ in visited_layers:
```

Comment:
In my understanding, are these lines only for an optimization, not for a bug fixing, and it looks any layers without weight could be ALLOWED_SHARED_KERAS_TYPES, right?

Reply:
The issue arose when I tied 2 LSTM layers together in `test_layers.py:test_rnn_state_passing`. Since there are 2 state outputs (hidden and cell) from the 1st LSTM to the 2nd LSTM, it winds up creating a duplicate object for the 1st LSTM and failing. This is because it does not correctly recognize that the 1st LSTM layer has already been considered, because the hidden output node is left over (not present in `visited`) when the cell output is traversed. I found it easier to fix the issue by ignoring any layer that had already been processed, but provided a mechanism to handle cases when this was desirable. I am not that familiar with this part of the code, so there may be a more optimal solution.

Reply:
This change will break mask_rcnn in existing nightly build, see [here](https://dev.azure.com/onnxmltools/ketone/_build/results?buildId=11028&view=logs&j=b5dc3593-0456-5043-06cc-36317977280a)

Reply:
Fixed [here](https://github.com/onnx/keras-onnx/pull/427)